sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -97,6 +97,14 @@ object DateTimeUtils {
     millisLocal - getOffsetFromLocalMillis(millisLocal, timeZone)
   }
 
+  def microsToDays(epochMicros: SQLTimestamp): SQLDate = {
+    Math.floorDiv(epochMicros, MICROS_PER_DAY).toInt
+  }
+
+  def daysToMicros(days: SQLDate): SQLTimestamp = {
+    Math.multiplyExact(days, MICROS_PER_DAY)
```

Comment:
shall we use `MICROSECONDS.toDay`?

Reply:
Yes, we can use `MICROSECONDS.toDay` is we do believe that `days` are in a reasonable range, and `days * 86400000000L` don't overflow `Long`.

Reply:
by SQL standard, max date value is `9999-12-31`, I think it won't overflow?